src/parser.ts

```diff
@@ -172,6 +178,7 @@ export function createAnsiSequenceParser() {
             foreground,
             background,
             decorations: new Set(decorations),
+            offset: position,
```

Comment:
This will be the position of the current parse call, but not the beginning of the full input if it is being parsed in chunks. I think the offset should be maintained at the parser level